src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -850,6 +1197,137 @@ EmberEventControl * ColorControlServer::configureHSVEventControl(EndpointId endp
     return controller;
 }
 
+/**
+ * @brief executes move to saturation command
+ *
+ * @param saturation target saturation
+ * @param transitionTime transition time in 10th of seconds
+ * @param endpoint target endpoint where to execute move
+ * @return Status::Success if successful,Status::UnsupportedEndpoint if the saturation transition state doesn't exist,
+ * Status::ConstraintError if the saturation is above maximum
+ */
+Status ColorControlServer::moveToSaturation(uint8_t saturation, uint16_t transitionTime, EndpointId endpoint)
+{
+    Color16uTransitionState * colorSaturationTransitionState = getSaturationTransitionState(endpoint);
+    VerifyOrReturnError(nullptr != colorSaturationTransitionState, Status::UnsupportedEndpoint);
+
+    if (transitionTime == 0)
+    {
+        transitionTime++;
+    }
+
+    // New command.  Need to stop any active transitions.
+    stopAllColorTransitions(endpoint);
+
+    // Handle color mode transition, if necessary.
+    handleModeSwitch(endpoint, COLOR_MODE_HSV);
+
+    // now, kick off the state machine.
+    initSaturationTransitionState(endpoint, colorSaturationTransitionState);
+    colorSaturationTransitionState->finalValue     = saturation;
+    colorSaturationTransitionState->stepsRemaining = transitionTime;
+    colorSaturationTransitionState->stepsTotal     = transitionTime;
+    colorSaturationTransitionState->endpoint       = endpoint;
```

Comment:
Where did the setting of `colorSaturationTransitionState->timeRemaining` go?

Please double-check all these refactored functions carefully and document in the PR any changes to them that were not just code movement and the reasons for the changes.

Reply:
Reverted unintentional changes, only change kept was the order in which `!shouldExecuteIfOff`. This was done so the new "move to" helpers have a coherent behaviour with the already existing moveToColorTemp.